setup.cfg

```diff
@@ -1,2 +1,47 @@
 [metadata]
 license_file = LICENSE.txt
+
+[flake8]
+ignore =
+    # E12X are indentatitons for bracketing
+    E12,
+    # E2XX are whitespaces around operators
+    E2,
+    # E3XX blank lines
+    E3,
+    # E402 module level import not at top of file
+    E402,
+    # E501 line too long (81 > 79 characters) (autopep8 catches this)
+    E501,
+    # E502 the backslash is redundant between brackets
+    E502,
+    # W503 line break before binary operator
+    W503,
+    # E712 comparison to False should be 'if cond is False:' or 'if not cond:'
+    # The reason this one is disable is that for arrays, it is useful to write
+    # array == False to return an array.
+    # you could also use np.logical_not but it is a little more verbose
+    E712,
+    # E741 ambiguous variable name 'l'
+    # some algorithms seem to use the variable l, like lower('L')
+    E741,
+    # F841 local variable 'ic' is assigned to but never used
+    # Unfortunately, I think some matplotlib calls are stateful
+    # therefore it is useful to ignore these warnings, at least for now
+    F841,
+    # F401 'package' imported but unused. This shows up too much in __init__ files
+    F401,
+    # F403 'from .manage_plugins import *' used; unable to detect undefined names
+    F403,
+    # W391 blank line at end of file
+    W391
+
+exclude =
+    test_*,
```

Comment:
why excluding tests?

Reply:
Tests often include many data arrays, that are well justified not to follow pep8.

Reply:
@hmaarrfk this is true, but test functions themselves should follow PEP8. I'd prefer finer-grained control here. Maybe it's not doable but I prefer to ignore these warnings rather than silence them completely.

Reply:
I really believe that warnings should be set to 0, except when they would add more noise (like in examples).

The idea is that currently the flake8 stuff is just fully being ignore.